.github/workflows/build-site.yaml

```diff
@@ -21,30 +21,35 @@ jobs:
 
     steps:
       # for debugging
-      - name: Print contexts
-        uses: crazy-max/ghaction-dump-context@v1
+      - uses: crazy-max/ghaction-dump-context@v2
```

Comment:
I might be misunderstanding how composite actions work, but in this case I don't think they would have much benefit in the case of `ghaction-dump-context`. It would just result in another workflow file for "dump-context", and then you'd just be calling it in all the same places like `uses: ../dump-context.yaml`?

Looking over other steps and workflows... I don't think there's _that_ much opportunity to remove repetition, or at least not enough benefit to justify the cost to the user. We could abstract out for example `ruby/setup-ruby@v1.172.0` into one, but that adds another workflow file to the already unruly list of them (which is already as minimal as I could figure out how to make it), and only saves a few lines and 2 repetitions of the ruby version. Since this affects difficulty of updating the user's template version (resolving merge conflicts), and because the user has to go through workflows frequently to find the proper one to run or inspect the logs of, I want to avoid adding new ones as much as possible. 

If you can think of a specific use of composite that makes things cleaner, please post it here, or open an issue if you think of it later.

Reply:
I might have mistaken the pattern, it looked like there were about 3-4 steps which repeated in several of the `.yml` files. The composite might save the work of having to retype the version specification for Ruby, among other options. Especially for the Ruby specification, it could help to reassure that everything is operating within the same environment expectations without needing to touch multiple files (as though through environment management).

I think the steps would be something like (very rough understanding):

```yaml
    steps:
      # for debugging
      - uses: crazy-max/ghaction-dump-context@v2

      - name: Checkout branch contents
        uses: actions/checkout@v4

      - name: Install Ruby packages
        uses: ruby/setup-ruby@v1.172.0
        with:
          ruby-version: "3.1"
          bundler-cache: true

      # for debugging
      - if: runner.debug == '1'
        uses: mxschmitt/action-tmate@v3
```

Reply:
![chrome_po0e9CObUv](https://github.com/greenelab/lab-website-template/assets/8326331/fbc3ea70-e2cb-4a2a-80d2-8e4cf4bfd87a)

Looking through the workflow steps, the problem is there are slight differences in parameters and orders. Perhaps that could be generalized with a composite action that takes a bunch of input parameters, but this harkens back to our discussion about making a function do too many things vs just repeating it.

Also, to investigate further what the steps look like, I made [this diagram](https://jsfiddle.net/pgeus8rb/show). There seems to be very little opportunity for reducing repetition. There are 41 total steps across the workflows, and 33 unique ones.

![image](https://github.com/greenelab/lab-website-template/assets/8326331/983346f3-1786-471e-b9b0-71a31211fc58)

Going to keep things as they are for now.